src/systems/audio-system.js

```diff
@@ -0,0 +1,78 @@
+  /**
+   *  workaround for: https://bugs.chromium.org/p/chromium/issues/detail?id=687574
+   *  1. grab the GainNode from the scene's THREE.AudioListener
+   *  2. disconnect the GainNode from the AudioDestinationNode (basically the audio out), this prevents hearing the audio twice.
+   *  3. create a local webrtc connection between two RTCPeerConnections (see this example: https://webrtc.github.io/samples/src/content/peerconnection/pc1/)
+   *  4. create a new MediaStreamDestination from the scene's THREE.AudioContext and connect the GainNode to it.
+   *  5. add the MediaStreamDestination's track  to one of those RTCPeerConnections
+   *  6. connect the other RTCPeerConnection's stream to a new audio element.
+   *  All audio is now routed through Chrome's audio mixer, thus enabling AEC, while preserving all the audio processing that was performed via the WebAudio API.
+   */
+
+  const audioEl = new Audio();
+  audioEl.setAttribute("autoplay", "autoplay");
+  audioEl.setAttribute("playsinline", "playsinline");
+
+  const context = THREE.AudioContext.getContext();
+  const loopbackDestination = context.createMediaStreamDestination();
+  const outboundPeerConnection = new RTCPeerConnection();
+  const inboundPeerConnection = new RTCPeerConnection();
+
+  const onError = e => {
+    console.error("RTCPeerConnection loopback initialization error", e);
+  };
+
+  outboundPeerConnection.addEventListener("icecandidate", e => {
+    inboundPeerConnection.addIceCandidate(e.candidate).catch(onError);
+  });
+
+  inboundPeerConnection.addEventListener("icecandidate", e => {
+    outboundPeerConnection.addIceCandidate(e.candidate).catch(onError);
+  });
+
+  inboundPeerConnection.addEventListener("track", e => {
+    audioEl.srcObject = e.streams[0];
+  });
+
+  gainNode.disconnect();
+  gainNode.connect(loopbackDestination);
+
+  loopbackDestination.stream.getTracks().forEach(track => {
+    outboundPeerConnection.addTrack(track, loopbackDestination.stream);
+  });
+
+  outboundPeerConnection.createOffer().then(offer => {
+    outboundPeerConnection.setLocalDescription(offer).catch(onError);
+
+    inboundPeerConnection
+      .setRemoteDescription(offer)
+      .then(() => {
+        inboundPeerConnection
+          .createAnswer()
+          .then(answer => {
+            inboundPeerConnection.setLocalDescription(answer).catch(onError);
+
+            outboundPeerConnection.setRemoteDescription(answer).catch(onError);
+          })
+          .catch(onError);
+      })
+      .catch(onError);
+  });
+}
+
+export class AudioSystem {
+  constructor(sceneEl) {
+    sceneEl.audioListener = sceneEl.audioListener || new THREE.AudioListener();
+    if (sceneEl.camera) {
+      sceneEl.camera.add(sceneEl.audioListener);
+    }
+    sceneEl.addEventListener("camera-set-active", evt => {
+      evt.detail.cameraEl.getObject3D("camera").add(sceneEl.audioListener);
+    });
+
+    if (!AFRAME.utils.device.isMobile() && /chrome/i.test(navigator.userAgent)) {
```

Comment:
Might want to explicitly exclude Oculus Browser here, since they also include "Chrome" in their UA and future versions will default to desktop mode.

Reply:
is this not needed on OculusBrowser?

Reply:
I don't think we've repro'd echo cancellation issues on Oculus Browser.

Reply:
I've heard anecdotally that there have been echo issues on Oculus Browser, so I plan on testing that a bit today and seeing if this PR helps with that or not. But otherwise, doesn't checking `!AFRAME.utils.device.isMobile()` effectively exclude oculus browser?

Reply:
`isMobile` might stop working on Quest if they switch their browser to desktop mode by default, which they are currently rolling out in trials.

Reply:
After doing some testing, it doesn't look like `isMobile` returns true in Oculus Browser right now- but it also looks like hardware AEC doesn't work on Quest. So we want to enable this feature on Oculus Browser anyways (but keep it disabled on true mobile browsers e.g. android chrome because they should be using hardware AEC).

Reply:
Well, turns out this workaround doesn't seem to work on Oculus Browser, so going to change to 
```
    if (/chrome/i.test(navigator.userAgent) && !/OculusBrowser/i.test(window.navigator.userAgent)) {
```

Will come back to this in the future and figure out why it's not working.

Reply:
would be interessting if AEC works in OCulus go ( different Hardware) maybe @Artyom17 has more infos on this?